packages/playwright/src/plugins/gitCommitInfoPlugin.ts

```diff
@@ -59,28 +59,54 @@ function linksFromEnv(): Pick<GitCommitInfo, 'revision.link' | 'ci.link'> {
     out['revision.link'] = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/commit/${process.env.GITHUB_SHA}`;
   if (process.env.GITHUB_SERVER_URL && process.env.GITHUB_REPOSITORY && process.env.GITHUB_RUN_ID)
     out['ci.link'] = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`;
+  if (process.env.GITHUB_REF_NAME && process.env.GITHUB_REF_NAME.endsWith('/merge')) {
+    const pullId = process.env.GITHUB_REF_NAME.substring(0, process.env.GITHUB_REF_NAME.indexOf('/merge'));
+    out['pull.link'] = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/pull/${pullId}`;
+    out['pull.base'] = process.env.GITHUB_BASE_REF;
+  }
   return out;
 }
 
-async function gitStatusFromCLI(gitDir: string): Promise<GitCommitInfo | undefined> {
+async function gitStatusFromCLI(gitDir: string, envInfo: Pick<GitCommitInfo, 'pull.base'>): Promise<GitCommitInfo | undefined> {
   const separator = `:${createGuid().slice(0, 4)}:`;
-  const { code, stdout } = await spawnAsync(
+  const commitInfoResult = await spawnAsync(
       'git',
       ['show', '-s', `--format=%H${separator}%s${separator}%an${separator}%ae${separator}%ct`, 'HEAD'],
       { stdio: 'pipe', cwd: gitDir, timeout: GIT_OPERATIONS_TIMEOUT_MS }
   );
-  if (code)
+  if (commitInfoResult.code)
     return;
-  const showOutput = stdout.trim();
+  const showOutput = commitInfoResult.stdout.trim();
   const [id, subject, author, email, rawTimestamp] = showOutput.split(separator);
   let timestamp: number = Number.parseInt(rawTimestamp, 10);
   timestamp = Number.isInteger(timestamp) ? timestamp * 1000 : 0;
 
-  return {
+  const result: GitCommitInfo = {
     'revision.id': id,
     'revision.author': author,
     'revision.email': email,
     'revision.subject': subject,
     'revision.timestamp': timestamp,
   };
+
+  const diffLimit = 1_000_000; // 1MB
+  if (envInfo['pull.base']) {
+    const pullDiffResult = await spawnAsync(
+        'git',
+        ['diff', envInfo['pull.base']],
+        { stdio: 'pipe', cwd: gitDir, timeout: GIT_OPERATIONS_TIMEOUT_MS }
+    );
+    if (!pullDiffResult.code)
+      result['pull.diff'] = pullDiffResult.stdout.substring(0, diffLimit);
```

Comment:
This is fine, but do we know how `substring` will perform when given very large amounts of text?

Additionally, if the truncation happens the last section of the diff is likely invalid, so if you were to attempt to apply it (idk why, but we are providing the data) weird things would happen.

Reply:
> This is fine, but do we know how substring will perform when given very large amounts of text?

I hope it's alright!

> Additionally, if the truncation happens the last section of the diff is likely invalid, so if you were to attempt to apply it (idk why, but we are providing the data) weird things would happen.

This isn't meant to apply, so I think that's fine.